pym/bob/cmds/build/builder.py

```diff
@@ -104,6 +108,100 @@ class RestartBuildException(Exception):
 class CancelBuildException(Exception):
     pass
 
+class JobServer:
+    def jobserver_thread(self):
+        while self.__running:
+            os.write(self.__w_server, os.read(self.__r_server, 1))
+
+    def __init__(self, jobs):
+        self.__running = True
+        self.thread = threading.Thread(target=self.jobserver_thread, args =())
+        self.__r_make, self.__w_server = os.pipe()
+        self.__r_server, self.__w_make = os.pipe()
+        os.write(self.__w_server, bytes(jobs))
+        self.thread.start()
+
+    def getMakeFd(self):
+        return [ self.__r_make, self.__w_make ]
+
+    def stop(self):
+        self.__running = False
+        os.write(self.__w_make, bytes(0x30))
+        self.thread.join()
+
+
+class JobServerSemaphore:
+    def __init__(self, value, fds):
+        if sys.platform == "win32":
```

Comment:
I would make the distinction in `Builder.cook()` on Windows we create the `asyncio.BoundedSemaphore ` as before, otherwise the new `JobServerSemaphore`. Then this class here does not need to make this distinction and will get smaller.